src/_zkapauthorizer/replicate.py

```diff
@@ -78,3 +117,133 @@ async def setup_tahoe_lafs_replication(client: Tahoe) -> Awaitable[str]:
 
     # Return the read-cap
     return rocap
+
+
+def with_replication(connection: Connection):
+    """
+    Wrap a replicating support layer around the given connection.
+    """
+    return _ReplicationCapableConnection(connection)
+
+
+@define
+class _ReplicationCapableConnection:
+    """
+    Wrap a ``sqlite3.Connection`` to provide additional snapshot- and
+    streaming replication-related features.
+
+    All of this type's methods are intended to behave the same way as
+    ``sqlite3.Connection``\ 's methods except they may also add some
+    additional functionality to support replication.
+    """
+
+    _conn: Connection
+
+    def snapshot(self) -> bytes:
+        """
+        Create and return a byte string representing a consistent, self-contained
+        snapshot of the wrapped database.
+        """
+        return snapshot(self._conn)
+
+    def close(self):
+        return self._conn.close()
+
+    def __enter__(self):
+        return self._conn.__enter__()
+
+    def __exit__(self, *args):
+        return self._conn.__exit__(*args)
+
+    def cursor(self):
+        return _ReplicationCapableCursor(self._conn.cursor())
+
+
+@define
+class _ReplicationCapableCursor:
+    """
+    Wrap a ``sqlite3.Cursor`` to provide additional streaming
+    replication-related features.
+
+    All of this type's attributes and methods are intended to behave the same
+    way as ``sqlite3.Cursor``\ 's methods except they may also add some
+    additional functionality to support replication.
+    """
+
+    _cursor: Cursor
+
+    @property
+    def lastrowid(self):
+        return self._cursor.lastrowid
+
+    @property
+    def rowcount(self):
+        return self._cursor.rowcount
+
+    def close(self):
+        return self._cursor.close()
+
+    def execute(self, statement, row=None):
+        if row is None:
+            args = (statement,)
+        else:
+            args = (statement, row)
+        self._cursor.execute(*args)
+
+    def fetchall(self):
+        return self._cursor.fetchall()
+
+    def fetchmany(self, n):
+        return self._cursor.fetchmany(n)
+
+    def fetchone(self):
+        return self._cursor.fetchone()
+
+    def executemany(self, statement, rows):
+        self._cursor.executemany(statement, rows)
+
+
+def netstring(bs: bytes) -> bytes:
```

Comment:
`netstring` and `statements_to_snapshot` just moved here from `test_recover.py`.  `connection_to_statements` is the old `snapshot` but with newlines removed because we do netstrings now.